### DISCUSSION TITLE
About the issue of custom window drag and drop

### DISCUSSION THREAD

**Discussion by KCKT0112** [AvaloniaUI/Avalonia#15749] [Q&A] [answered]
Hello everyone. I may have encountered some issues, but I cannot confirm if it is a bug, so I would like to have some discussion here.

I added a custom Titlbar to the window by drawing it myself, and it currently works well under MainWindow

Just like this:
https://github.com/LiuYunPlayer/TuneLab/blob/master/TuneLab/Views/MainWindow.axaml
https://github.com/LiuYunPlayer/TuneLab/blob/master/TuneLab/Views/MainWindow.axaml.cs

But I also used the same method to create a custom title bar in the sub window, but this time I couldn't drag it anyway. I set the IsHitTestVisible="False" property as needed, but it still cannot be dragged. According to the survey results, it was my custom title bar that blocked the drag and drop judgment area. However, it works normally in MainWindow, so I want to know if anyone can answer?

SubWindow：
https://github.com/LiuYunPlayer/TuneLab/blob/master/TuneLab/Views/LyricInput.axaml
https://github.com/LiuYunPlayer/TuneLab/blob/master/TuneLab/Views/LyricInput.axaml.cs

Thank you very much for your help.

**Comment by timunie** [comment, marked as answer]
I think `TitleBar.Background = Style.BACK.ToBrush();` this gives a background to the grid which then becomes hit-testable. 

> [!TIP]
> Use DevTools to view where events are eaten up.

**Reply by KCKT0112** [reply]
Thank you very much! I finally found the problem. Now it can drag normally!
